README.md

```diff
@@ -1,133 +1,111 @@
+<script>
+target.setAttribute('rendersubtree', ''); // makes #target render
+</script>
+```
+This div's subtree is not rendered (but the div itself is; this allows the div to show fallback or "loading..." affordances), and there is no need for the browser to do any rendering lifecycle phases for the subtree of the div. Custom element upgrades are not performed and resources are not loaded. The div lays out as if it had a single 200px by 200px child, which serves as a placeholder in order to take up the approximate layout size of the div's subtree. This allows page layout to be approximately correct, and preserves layout overflow size for scrolling. The browser may *not* render the content, even via a user-agent feature.
+
+```html
+<div rendersubtree="invisible activatable"  style="content-size: 200px 200px">...content</div>
+```
+Same as above, except that user-agent features may change the `rendersubtree` attribute to the empty string, causing the div's subtree to get rendered. More on this on ["Element activation by the user agent"](https://github.com/rakina/display-locking#element-activation-by-the-user-agent)
+
+```html
+<div rendersubtree style="content-size: 200px 200px">...content</div>
+```
+This div and its subtree render, but still has `style` and `layout` containment. `content-size` has no effect because it doesn't have `size` containment. This existence of `rendersubtree` is valuable beause when the div later becomes invisible, invalidations of rendering state are minimized.
+
+```html
+<div id=target rendersubtree="invisible" style="content-size: 200px 200px">...content...</div>
+<script>
+target.updateRendering().then(() => console.log(target.firstElementChild.offsetTop)); // fast!
+</script>
+```
+The div's subtree does not render to the screen, but when updateRendering is called, the browser does work in the background to prepare to render it quickly in the future. This includes loading external resources referred to in the subtree and custom element upgrades. It also may include running the rendering lifecycle steps for the subtree up to and including style, layout, paint and raster. When the returned promise resolves, reading layout or style-inducing properties on the subtree is expected to be fast. Changing the `rendersubtree` value to `visible` should also be fast.
+
+## Element activation by the user agent
+
+When an element is not rendered because it's a part of a not-rendered subtree caused by `rendersubtree`, there are some actions in the page that might require the element (and its ancestors) to be rendered to work properly. If all of the element's rendering ancestor whose `rendersubtree` attribute contains `invisible` also contains `activatable`, then the user agent can *activate* the element. This will change of the non-null rendersubtree value of all of its rendering ancestors to the empty string - causing the element and its ancestors to get rendered.
```

Comment:
```suggestion
When an element is not rendered because it's a part of a not-rendered subtree caused by `rendersubtree`, there are some actions in the page that might require the element (and its ancestors) to be rendered to work properly. If all of the element's rendering ancestors' `rendersubtree` attributes contain contain `activatable`, then the user agent can *activate* the element. This will change of the non-null rendersubtree value of all of its rendering ancestors to the empty string - causing the element and its ancestors to get rendered.
```